src/Swift.Bindings/tests/CryptoKit/Foundation.Source.cs

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Reflection;
+
+namespace Test
+{
+    /// <summary>
+    /// Represents a Swift type in C#.
+    /// </summary>
+    public unsafe interface ISwiftObject
+    {
+        public static abstract void* Metadata { get; }
```

Comment:
For the future - I'd really like to see Metadata be its own type - a struct containing either an IntPtr, a NativeHandle, or a void*. There will be sufficient funcs that will have Metadata as arguments that we are likely to have signature conflicts. Let's open up an issue for this.

Reply:
Created a tracking issue: https://github.com/dotnet/runtimelab/issues/2705